SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -602,46 +574,51 @@ void FSpatialGDKEditorToolbarModule::VerifyAndStartDeployment()
+		USpatialOSSingleWorkerStrategy* DefaultStrategy = USpatialOSSingleWorkerStrategy::StaticClass()->GetDefaultObject<USpatialOSSingleWorkerStrategy>();
+		USpatialOSAbstractLoadBalancingStrategy* LoadBalancingStrat = DefaultStrategy;
+
 		if (SpatialGDKSettings->bEnableUnrealLoadBalancer)
 		{
-			FIntPoint WorldDimensions;
-			FWorkerTypeLaunchSection WorkerLaunch;
-
-			if (FillWorkerLaunchConfigFromWorldSettings(*EditorWorld, WorkerLaunch, WorldDimensions))
+			if (GetLoadBalancingStrategyFromWorldSettings(*EditorWorld, LoadBalancingStrat, LaunchConfigDescription.World.Dimensions))
 			{
-				LaunchConfigDescription.World.Dimensions = WorldDimensions;
-				LaunchConfigDescription.ServerWorkers.Empty(SpatialGDKSettings->ServerWorkerTypes.Num());
-
-				for (auto WorkerType : SpatialGDKSettings->ServerWorkerTypes)
-				{
-					LaunchConfigDescription.ServerWorkers.Add(WorkerLaunch);
-					LaunchConfigDescription.ServerWorkers.Last().WorkerTypeName = WorkerType;
-				}
+				LoadBalancingStrat->AddToRoot();
 			}
 		}
 
-		for (auto& WorkerLaunchSection : LaunchConfigDescription.ServerWorkers)
+		TMap<FName, FWorkerLaunchConfiguration> WorkersMap;
+
+		for (auto& WorkerLaunchSection : SpatialGDKSettings->ServerWorkerDesc)
 		{
-			WorkerLaunchSection.bManualWorkerConnectionOnly = true;
+			FWorkerLaunchConfiguration WorkerConf;
+			WorkerConf.WorkerConfig = WorkerLaunchSection.Value;
+			WorkerConf.WorkerLoadBalancing = LoadBalancingStrat;
+			WorkerConf.WorkerConfig.bManualWorkerConnectionOnly = true;
+			WorkersMap.Add(WorkerLaunchSection.Key, WorkerConf);
 		}
 
-		if (!ValidateGeneratedLaunchConfig(LaunchConfigDescription))
+		if (!ValidateGeneratedLaunchConfig(LaunchConfigDescription, WorkersMap))
 		{
 			return;
 		}
 
-		GenerateDefaultLaunchConfig(LaunchConfig, &LaunchConfigDescription);
-		LaunchConfigDescription.SetLevelEditorPlaySettingsWorkerTypes();
+		GenerateDefaultLaunchConfig(LaunchConfig, &LaunchConfigDescription, WorkersMap);
+		SetLevelEditorPlaySettingsWorkerTypes(WorkersMap);
 
 		// Also create default launch config for cloud deployments.
 		{
-			for (auto& WorkerLaunchSection : LaunchConfigDescription.ServerWorkers)
+			for (auto& WorkerLaunchSection : WorkersMap)
 			{
-				WorkerLaunchSection.bManualWorkerConnectionOnly = false;
+				WorkerLaunchSection.Value.WorkerConfig.bManualWorkerConnectionOnly = false;
 			}
 
 			FString CloudLaunchConfig = FPaths::Combine(FPaths::ConvertRelativePathToFull(FPaths::ProjectIntermediateDir()), FString::Printf(TEXT("Improbable/%s_CloudLaunchConfig.json"), *EditorWorld->GetMapName()));
-			GenerateDefaultLaunchConfig(CloudLaunchConfig, &LaunchConfigDescription);
+			GenerateDefaultLaunchConfig(CloudLaunchConfig, &LaunchConfigDescription, WorkersMap);
+		}
+
+		// Manage LoadBalancingStrategy lifecycle.
+		if (LoadBalancingStrat != DefaultStrategy)
+		{
+			LoadBalancingStrat->RemoveFromRoot();
```

Comment:
what does this line do?

Reply:
Allow the object to be garbage collected.